### PR TITLE
Assemble multiple PDUs from a single buffer

### PR DESCRIPTION
**Pull request by TomerShalish**

### CODE REVIEW

**Comment by geynis** [comment]
you should test that the TCP use case is not affected. seems maybe for tcp you need to give the single pdu. line 185

**Comment by TomerShalish** [comment]
@geynis Do you have any leftover pcaps from May? I am lazy, and would rather not create any myself `:D

**Comment by geynis** [comment]
please rebase over develop branch

File: someip.lua [hunk 1]
```diff
@@ -154,13 +162,20 @@ local function someip_pdu_dissect(buf,pinfo,root)
     if (buf(0,4):uint() == 0xffff8100) and (buf:len() > SOMEIP_HDR_LENGTH)  then
         Dissector.get("sd"):call(buf(SOMEIP_HDR_LENGTH):tvb(),pinfo,root)
     elseif (buf:len() > SOMEIP_HDR_LENGTH) then
-        Dissector.get("data"):call(buf(SOMEIP_HDR_LENGTH):tvb(),pinfo,root)
+        Dissector.get("data"):call(buf(SOMEIP_HDR_LENGTH,-SOMEIP_HDR_LENGTH+8+get_someip_length(buf,pinfo,0)):tvb(),pinfo,root)
     end
 
+    -- Return next SOMEIP packet --
```
Comment (geynis) [inline someip.lua]: you should redesign, such that this dissector will call itself as a new protocol layer.

**Review by geynis — APPROVED** [review]

**Pull request merged**